feat: add support for multi-arch repositories

gaige opened the pull request:
- added support for multi-arch repositories
- manually set os to ‘linux’ as there are (at least for now) no ‘smartos’ docker images

gaige committed:
feat: add support for multi-arch repositories

- added support for multi-arch repositories
- manually set os to ‘linux’ as there are (at least for now) no ‘smartos’ docker images

gaige committed:
fix: update tests for single arch result

gaige committed:
docs: update comments

danmcd commented:
See TritonDataCenter/smartos-live#1143 in case it helps.

cneira commented:
@gaige, I added your work in https://github.com/TritonDataCenter/node-docker-registry-client/pull/45. I added a couple of changes, but all tests are passing now(let me know if you are ok with that) I added you as the author and myself as the coauthor.

gaige commented:
> @gaige, I added your work in #45. I added a couple of changes, but all tests are passing now(let me know if you are ok with that) I added you as the author and myself as the coauthor.

Works for me. I'm all about the fastest way to get this into the code base :-)
